WordPress/gutenberg
# [RNMobile] Upgrade to RN 0.66

## Related PRs

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/4235
* https://github.com/wordpress-mobile/WordPress-Android/pull/15566
* https://github.com/wordpress-mobile/WordPress-iOS/pull/17594

<details><summary>Third-Party Fork PRs</summary>
 
* https://github.com/wordpress-mobile/react-native-masked-view/pull/6
* https://github.com/wordpress-mobile/react-native-slider/pull/10
* https://github.com/wordpress-mobile/react-native-gesture-handler/pull/8
* https://github.com/wordpress-mobile/react-native-get-random-values/pull/5
* https://github.com/wordpress-mobile/react-native-hsv-color-picker/pull/6
* https://github.com/wordpress-mobile/react-native-linear-gradient/pull/5
* https://github.com/wordpress-mobile/react-native-prompt-android/pull/7
* https://github.com/wordpress-mobile/react-native-reanimated/pull/6
* https://github.com/wordpress-mobile/react-native-safe-area-context/pull/6
* https://github.com/wordpress-mobile/react-native-screens/pull/6
* https://github.com/wordpress-mobile/react-native-svg/pull/16
* https://github.com/wordpress-mobile/react-native-video/pull/14
* https://github.com/wordpress-mobile/react-native-webview/pull/4

</details>

## Description
Upgrade React Native to 0.66. See the [React Native Upgrade Helper](https://href.li/?https://react-native-community.github.io/upgrade-helper/?from=0.64.0&to=0.66.2) for additional details. 

Fixes #34654. Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/4219. 

## How has this been tested?

Performed of Writing Flow and UBE test cases. 

<details><summary>Help & Support</summary>

1. Launch the iOS editor. 
2. Tap the three-dot button in the top-right corner. 
3. Tap _Help & Support_.
4. Tap one of the help articles. 
5. Tap _Back_. 
6. Tap _Close_. 
6. ℹ️ **Expected:** The app does not freeze and is able to scroll and add/edit/remove blocks. 

</details>

<details><summary>Link Picker</summary>

1. Launch the iOS editor.
2. Add a Paragraph block with text. 
3. Perform various linking actions multiple times over closing the block settings bottom sheet throughout. 
    - Add a link _without_ text selection. 
    - Add link _with_ text selection. 
    - Add link with clipboard URL. 
    - Modify existing link text. 
    - Modify existing link URL. 
    - Remove existing link with _Remove link_ button. 
    - Remove existing link by clearing URL within link picker.  
    - Remove existing link by clearing link text. 
4. ℹ️ **Expected:** The app does not freeze and is able to scroll and add/edit/remove blocks. 

</details>

<details><summary>Cover Focal Point</summary>

1. Launch the iOS editor. 
2. Add a Cover block and set an image as its media. 
3. Open the Cover block settings. 
4. Tap _Edit focal point_. 
5. Tap _Cancel_. 
6. Close the Cover block settings. 
6. ℹ️ **Expected:** The app does not freeze and is able to scroll and add/edit/remove blocks. 

</details>

<details><summary>Cover Opacity</summary>

1. Launch the iOS editor. 
2. Add a Cover block and set an image as its media. 
3. Open the Cover block settings. 
4. Modify the Opacity by dragging the slider. 
6. Close the Cover block settings. 
6. ℹ️ **Expected:** The app does not freeze and is able to scroll and add/edit/remove blocks. 

</details>

## Screenshots 
n/a

## Types of changes
Chore

## Checklist:
